src/MovingAverage.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * Moving Average Filter (linear FIR filter with equal weights)
+ */
+#include "iarc7_sensors/MovingAverage.hpp"
+
+#include <ros/ros.h>
+
+namespace iarc7_sensors {
+
+MovingAverage::MovingAverage(const ros::Duration& time) : filter_time_constant_(time) {}
+
+void MovingAverage::updateFilter(double value, double derivative, const ros::Time& time) {
```

Comment:
About the derivative, I agree we should use it, I was thinking we would merge in the current state and add that later.  It'll help with extrapolation and with getting rid of lag.  The timestamp weight handler might be taking it a little further than we want to for this project.  I guess you mean a generic linear FIR filter that doesn't necessarily have equal weights?